app/models/manageiq/providers/hawkular/middleware_manager/alert_manager.rb

```diff
@@ -105,15 +105,21 @@ def convert_to_group_conditions(miq_alert)
     end
 
     def mw_server_metrics_by_column
-      MiddlewareServer.live_metrics_config['middleware_server']['supported_metrics_by_column']
+      MiddlewareServer.live_metrics_config['supported_metrics']['default'].invert
     end
 
     def mw_datasource_metrics_by_column
-      MiddlewareDatasource.live_metrics_config['middleware_datasource']['supported_metrics_by_column']
+      MiddlewareDatasource.live_metrics_config['supported_metrics']['default'].invert
     end
 
-    def mw_messaging_topic_metrics_by_column
-      MiddlewareMessaging.live_metrics_config['middleware_messaging_jms_topic']['supported_metrics_by_column']
+    def mw_messaging_metrics_by_column
+      all = {}
+      supported_metrics = MiddlewareMessaging.live_metrics_config['supported_metrics']
+      supported_metrics.keys.each { |resource_type|
+        type_metrics = supported_metrics[resource_type]
+        all.merge!( supported_metrics[resource_type].invert ) if type_metrics
```

Comment:
Isn't a little better:?
```ruby
all = type_metrics.invert if type_metrics

Reply:
If there are multiple types (and there are already 2) then we need to merge all of the mappings into one in order to return a master hash for lookup.  It's sort of clunky but trigger definition is rare.

maybe I missed your point but I think in your suggestion 'all' would get replaced for each type, and ultimately return just the last type's mappings.

Reply:
@jshaughn yes, you are right...
Perhaps, then, `all.merge!( type_metrics.invert ) if type_metrics`
but that's will just make it a little shorter. It's OK for me. I can merge as is if you don't want to change it.

Reply:
No, that will blow up if there are no type_metrics, that is why there is 'if type_metrics'.

Reply:
Indeed my suggestion has the conditional. But, anyway, I'll merge.

Reply:
oh, right, I'll change it in a future update.